### PR TITLE
Expose server error-text in array-UDF response [RFC]

### PR DESCRIPTION
**Pull request by johnkerl**
Goal: make API-response error messages clearer and more actionable.

Before:

```
tiledbcloud: received error response: API server error
```

After:

```
tiledbcloud: received error response:
Server returned 500 response status code. Content:
 {"code":5000,"message":"json: cannot unmarshal string into Go value of type []json.RawMessage","request_id":"6b67bcaa-5e8e-4a64-b940-001b394503a9"}
```

Maybe this should be:

```
tiledbcloud: received error response:
Server returned 500 response status code.
Message: json: cannot unmarshal string into Go value of type []json.RawMessage
```

Room for discussion (hence "RFC" in this PR's title):

* I think more error-text is better than less
  * If it's directly actionable by the user (not the case in the example above, admittedly) then they can iterate self-sufficiently
  * If it isn't, nonetheless a particulary worded error-text can help us give them better support
* Formatting might be improved, e.g. if we JSON-decode the response and present the user instead with the text I labeled above with _Maybe this should be ..._
* I did this at a single API callsite (see files on this PR)
  * I can apply the same hand-edits across many callsites
  * Or maybe I can make a helper function and apply _that_ across many callsites

### CODE REVIEW

File: R/execute.R [hunk 1]
```diff
@@ -110,10 +110,12 @@ execute_array_udf <- function(namespace, array, udf, layout=NULL, selectedRanges
   # object.
   if (typeof(resultObject) != "raw") {
     className <- class(resultObject)[1]
+    print(resultObject)
```
Comment (eddelbuettel) [inline R/execute.R:113]: Printing the entire object _may_ not be a good idea...  But it is hard to say.

File: R/execute.R [hunk 2]
```diff
@@ -110,10 +110,12 @@ execute_array_udf <- function(namespace, array, udf, layout=NULL, selectedRanges
...
+    print(str(resultObject))
```
Comment (eddelbuettel) [inline R/execute.R:114]: Calling `str()` strikes me as a bad idea; it is a somewhat 'internal' view.  Users may or may not be R programmers.

File: R/execute.R [hunk 3]
```diff
@@ -110,10 +110,12 @@ execute_array_udf <- function(namespace, array, udf, layout=NULL, selectedRanges
...
     if (className == "ApiResponse") {
-      stop(paste("tiledbcloud: received error response:", resultObject$content))
+      stop(paste("tiledbcloud: received error response:", resultObject$content, sep="\n"))
```
Comment (eddelbuettel) [inline R/execute.R:116]: Not sure if that was initially generated that way but `paste()` inside `stop()` is redundant as `stop()` already glues arguments together---try `stop("AB", "CD", "DE", call. = FALSE)`.

I am not sure the `\n` is a good idea either.  But that gets to tastes.

There is a handy base R function `strwrap(txt, width)` that reformats `txt` to `width`.  But then this an error printer, not a pretty printer.

**Review by eddelbuettel — COMMENTED** [review]
Some comments added inline.

File: R/execute.R (hunk 1 above)
Comment (johnkerl) [reply R/execute.R:113]: @eddelbuettel agreed

I like this better:

```
  bodyAsJSON <- rawToChar(resp$content)
  bodyParsed <- jsonlite::fromJSON(bodyAsJSON)
  ApiResponse$new(paste("Server returned" , httr::status_code(resp) , "response status code. Message:\n", 
    bodyParsed$message), resp)
```

giving

```
tiledbcloud: received error response:
Server returned 500 response status code. Message:
 json: cannot unmarshal string into Go value of type []json.RawMessage
```

File: R/execute.R (hunk 2 above)
Comment (johnkerl) [reply R/execute.R:114]: @eddelbuettel  good call; these were debug-prints which needn't stay

**Review by eddelbuettel — APPROVED** [review]
Overall looks fine methinks.

File: R/execute.R (hunk 3 above)
Comment (johnkerl) [reply R/execute.R:116]: Thanks @eddelbuettel !!

There are lots of API callsites ... I think I'll take the bits I wrote, put them in a helper function, then have the various API callsites call it on non-2xx cases ... then flip this PR from RFC to non-RFC

Thanks for the tips on `stop`/`paste`/etc! :) Your R expertise continues to be wonderful to draw from :)

**Comment by johnkerl** [comment]
Note: we need to address the `openssl` issue in CI (unrelated to this PR).

**Pull request merged**